src/back/spv/writer.rs

```diff
@@ -716,10 +709,28 @@ impl Writer {
     fn write_type_declaration_arena(
         &mut self,
         arena: &Arena<crate::Type>,
+        layouter: &Layouter,
         handle: Handle<crate::Type>,
     ) -> Result<Word, Error> {
         let ty = &arena[handle];
+
+        if let Some(local) = self.physical_layout.make_local(&ty.inner) {
+            match self.lookup_type.entry(LookupType::Local(local)) {
+                // if it's already known as local, re-use it
+                Entry::Occupied(e) => {
+                    let id = *e.into_mut();
+                    self.lookup_type.insert(LookupType::Handle(handle), id);
+                    return Ok(id);
+                }
+                // also register the type as "local", to avoid duplication
+                Entry::Vacant(e) => {
+                    let id = self.id_count + 1; // has to match the next ID
```

Comment:
Is there a reason to not use `self.generate_id()` here? Wouldn't this give a collision for the first next id that is generated via `self.generate_id()`, as `self.id_count` is not incremented here, but will be at line 732?

Reply:
fixed now with `IdGenerator` struct